pkg/apis/agent/v1alpha1/webhook.go

```diff
@@ -38,7 +38,7 @@ func (a *Agent) GetWarnings() []string {
 	if a == nil {
 		return nil
 	}
-	if len(a.Spec.PolicyID) == 0 {
+	if a.Spec.Mode == AgentFleetMode && len(a.Spec.PolicyID) == 0 {
```

Comment:
I think this change makes perfect sense, but I also think it's a perfect opportunity for a set of unit tests around this function.

Reply:
There are existing unit tests, will fix those and add a new one

Reply:
I was only looking for a `webhooks_test.go` file and didn't see it. My apologies.